packages/commonwealth/client/scripts/models/Thread.ts

```diff
@@ -267,179 +287,123 @@ export class Thread implements IUniqueId {
-    numberOfComments,
-    reactions,
-    reactionIds,
-    reactionType,
-    reactionTimestamps,
-    reactionWeights,
-    reaction_weights_sum,
-    addressesReacted,
-    reactedProfileName,
-    reactedProfileAvatarUrl,
-    reactedAddressLastActive,
-    canvas_signed_data,
-    canvas_msg_id,
-    links,
-    discord_meta,
-    userId,
-    user_id,
-    profile_name,
-    avatar_url,
-    address_last_active,
-    associatedReactions,
-    associatedContests,
-    recentComments,
-    ContestActions,
-    content_url,
-  }: {
-    marked_as_spam_at: string;
-    title: string;
-    body?: string;
-    id: number;
-    kind: ThreadKind;
-    stage: ThreadStage;
-    community_id: string;
-    url?: string;
-    pinned?: boolean;
-    links?: Link[];
-    canvas_signed_data?: string;
-    canvas_msg_id?: string;
-    collaborators?: any[];
-    last_edited: string;
-    locked_at: string;
-    last_commented_on: string;
-    created_at: string;
-    updated_at: string;
-    archived_at?: string;
-    read_only: boolean;
-    has_poll: boolean;
-    numberOfComments?: number;
-    topic: Topic;
-    reactions?: any[]; // TODO: fix type
-    reactionIds?: any[]; // TODO: fix type
-    addressesReacted?: any[]; //TODO: fix type,
-    reactedProfileName?: string[];
-    reactedProfileAvatarUrl?: string[];
-    reactedAddressLastActive?: string[];
-    reactionType?: any[]; // TODO: fix type
-    reactionTimestamps?: string[];
-    reactionWeights?: number[];
-    reaction_weights_sum: string;
-    ThreadVersionHistories: ThreadVersionHistory[];
-    Address: any; // TODO: fix type
-    discord_meta?: any;
-    userId: number;
-    user_id: number;
-    profile_name: string;
-    avatar_url: string;
-    address_last_active: string;
-    associatedReactions?: AssociatedReaction[];
-    associatedContests?: AssociatedContest[];
-    recentComments: RecentComment[];
-    ContestActions: ContestActionT[];
-    content_url: string | null;
-  }) {
-    this.author = Address?.address;
-    this.title = getDecodedString(title);
-    // @ts-expect-error StrictNullChecks
-    this.body = getDecodedString(body);
-    this.id = id;
-    this.identifier = `${id}`;
-    this.createdAt = moment(created_at);
-    this.updatedAt = moment(updated_at);
-    this.topic = { ...topic };
-    this.kind = kind;
-    this.stage = stage;
-    this.authorCommunity = Address?.community_id;
-    // @ts-expect-error StrictNullChecks
-    this.pinned = pinned;
-    // @ts-expect-error StrictNullChecks
-    this.url = url;
-    this.communityId = community_id;
-    this.readOnly = read_only;
-    this.collaborators = collaborators || [];
-    // @ts-expect-error StrictNullChecks
-    this.lastCommentedOn = last_commented_on ? moment(last_commented_on) : null;
-    this.hasPoll = has_poll;
-    // @ts-expect-error StrictNullChecks
-    this.lastEdited = last_edited
-      ? moment(last_edited)
+  constructor(
+    t: ThreadView & {
+      // TODO: fix other type variants
```

Comment:
should this todo be done in this PR? If no, let's create a ticket and link it here, otherwise this TODO is not actionable.

Reply:
Not in this PR, this is probably an incremental UI work that we can execute on tech-debt Fridays... feel free to create ticket or bucket, or let me know if you want me to create one for you

Reply:
because you were touching it recently and have more context, could you create it and describe the scope of work, please?